src/jltypes.c

```diff
@@ -3051,10 +3051,10 @@ void jl_init_types(void) JL_GC_DISABLED
                         jl_emptysvec, 0, 0, 2);
 
     jl_lineinfonode_type =
-        jl_new_datatype(jl_symbol("LineInfoNode"), core, jl_any_type, jl_emptysvec,
-                        jl_perm_symsvec(5, "module", "method", "file", "line", "inlined_at"),
-                        jl_svec(5, jl_module_type, jl_any_type, jl_symbol_type, jl_int32_type, jl_int32_type),
-                        jl_emptysvec, 0, 0, 5);
+        jl_new_datatype(jl_symbol("LegacyLineInfoNode"), core, jl_any_type, jl_emptysvec,
```

Comment:
I don't think we should rename this, especially if we are still using it. Maybe it can be merged with `LineNumberNode` though?

Reply:
We aren't really still using it though. It appears briefly in the output from lowering until we can wrap it in the output (also from lowering) of a CodeInfo, at which point we destroy this and allocate the forward edges structure--which becomes the only types the user should see now. It will probably make more sense later to have lowering emit the new format directly--at which point this placeholder type can be deleted the rest of the way

Reply:
Deleting it will be fine, but as long as it exists we should just leave it alone.

Reply:
It looks like we still needed this exact type in Core to support serialization of legacy code, so I also now define the user-accessible type in boot.jl directly, with the hope that we remove this object soon (as soon as we can teach lowering not to use it as an intermediate representation)